vexination/walker/src/lib.rs

```diff
@@ -1,70 +1,70 @@
-#![allow(dead_code)]
-#![allow(unused_variables)]
-#![allow(unused_imports)]
-use std::{
-    path::PathBuf,
-    process::ExitCode,
-    str::FromStr,
-    time::{Duration, SystemTime},
+use crate::scanner::{Options, Scanner};
+use anyhow::anyhow;
+use clap::{arg, command, Args};
+use std::path::PathBuf;
+use std::process::ExitCode;
+use std::sync::Arc;
+use std::time::SystemTime;
+use trustification_auth::client::{OpenIdTokenProviderConfig, OpenIdTokenProviderConfigArguments};
+use trustification_infrastructure::{
+    endpoint::{self, Endpoint},
+    Infrastructure, InfrastructureConfig,
 };
-
-use prometheus::Registry;
-use time::{Date, Month, UtcOffset};
-use trustification_auth::client::OpenIdTokenProviderConfigArguments;
-use trustification_infrastructure::{Infrastructure, InfrastructureConfig};
 use url::Url;
-use walker_common::validate::ValidationOptions;
+use walker_common::sender::provider::TokenProvider;
 
-mod server;
+mod scanner;
 
-#[derive(clap::Args, Debug)]
-#[command(about = "Run the api server", args_conflicts_with_subcommands = true)]
-pub struct Run {
-    /// Source URL or path
-    #[arg(short, long)]
-    pub(crate) source: String,
+const DEVMODE_SOURCE: &str = "https://www.redhat.com/.well-known/csaf/provider-metadata.json";
 
-    /// Vexination upload url
-    #[arg(short = 'S', long)]
-    pub(crate) sink: Url,
+#[derive(Args, Debug)]
+#[command(
+    about = "Run the VEX walker",
+    args_conflicts_with_subcommands = true,
+    rename_all_env = "SCREAMING_SNAKE_CASE"
+)]
+pub struct Run {
+    /// Apply reasonable settings for local development. Do not use in production!
+    #[arg(long = "devmode", default_value_t = false)]
+    pub devmode: bool,
 
-    /// Distributions to ignore
-    #[arg(long, default_value = "Vec::new()")]
-    ignore_distributions: Vec<Url>,
+    /// Long-running mode. The index file will be scanned for changes every interval.
+    #[arg(long = "scan-interval")]
+    pub scan_interval: Option<humantime::Duration>,
 
     /// OpenPGP policy date.
     #[arg(long)]
-    policy_date: Option<humantime::Timestamp>,
+    pub policy_date: Option<humantime::Timestamp>,
 
-    /// Enable OpenPGP v3 signatures. Conflicts with 'policy_date'.
-    #[arg(short = '3', long = "v3-signatures", conflicts_with = "policy_date")]
```

Comment:
That flag is an absolute must have. And explaining the "policy date" idea in the docs is something that I would really like to avoid.